core/templates/components/summary-tile/subtopic-summary-tile.component.ts

```diff
@@ -53,34 +52,37 @@ export class SubtopicSummaryTileComponent implements OnInit {
   openSubtopicPage(): void {
     // This component is being used in the topic editor as well and
     // we want to disable the linking in this case.
-    if (!this.classroomUrlFragment || !this.topicUrlFragment) {
+    let urlFragment = this.subtopic.getUrlFragment();
+    if (!this.classroomUrlFragment || !this.topicUrlFragment || !urlFragment) {
       return;
     }
     this.windowRef.nativeWindow.open(
       this.urlInterpolationService.interpolateUrl(
         TopicViewerDomainConstants.SUBTOPIC_VIEWER_URL_TEMPLATE, {
           classroom_url_fragment: this.classroomUrlFragment,
           topic_url_fragment: this.topicUrlFragment,
-          subtopic_url_fragment: this.subtopic.getUrlFragment()
+          subtopic_url_fragment: urlFragment
         }
       ), '_self'
     );
   }
 
   ngOnInit(): void {
-    this.thumbnailBgColor = this.subtopic.getThumbnailBgColor();
     this.subtopicTitle = this.subtopic.getTitle();
-    if (this.subtopic.getThumbnailFilename()) {
-      this.thumbnailUrl = (
-        this.assetsBackendApiService.getThumbnailUrlForPreview(
-          AppConstants.ENTITY_TYPE.TOPIC, this.topicId,
-          this.subtopic.getThumbnailFilename()));
-    } else {
-      this.thumbnailUrl = null;
```

Comment:
throw error if file not exist instead of saving url as null.

Reply:
Have you tested this locally?

Reply:
take reference from learner topic goal summary, ditto things we are doing.

But as per your [comment in conversation](https://github.com/oppia/oppia/pull/15620#discussion_r907128765), i also think we should mention its else condition, and what happen if else condition run.

Reply:
Okay intialize thumbnailUrl with null, then assign value in ngOnInit().
So default value of thumbnailUrl  is null, no need of else statement then.